x/mongo/driver/topology/connection_test.go

```diff
@@ -110,6 +110,19 @@ func TestConnection(t *testing.T) {
 				assert.NotNil(t, err, "expected connect error %v, got nil", want)
 				assert.Equal(t, want, got, "expected error %v, got %v", want, got)
 			})
+			t.Run("cancelConnectContext is nil after connect", func(t *testing.T) {
+				conn, got := newConnection(context.Background(), address.Address(""))
+
+				if got != nil {
```

Comment:
Use `assert.Nil` here instead. Also, if there is a case where you need to use an explicit conditional rather than an `assert` call, you should generally use `t.Fatalf` rather than `t.Errorf`. The difference is that `Errorf` internally calls `t.Fail`, which marks the test as failed but allows execution to continue. If there was an error, though, the value of `conn` is unknown (based on the code, we know it's nil, but best to treat it as unknown/undefined), so using it below would likely cause the test to panic. `Fatalf` internally calls `t.FailNow`, which both marks the test as failed and causes it to exit immediately without continuing execution.